### DISCUSSION TITLE
Deleting PDF pages

### DISCUSSION THREAD

**Discussion by tc4u** [espresso3389/pdfrx#205] [Q&A]
First, thank you for pdfrx! It's working well for me so far. 

My application displays PDFs but also allows the user to delete one or more pages. I tried using Pdfium thru ffi to delete a page and that works (confirmed by calling PDFium getpagecount afterward). 

But, the problem is that nothing on the pdfrx side knows about that change. It still tracks the original pages. I'm assuming it's because Dart is pass by value and my delete is only modifying a copy (ahh, the multiverse, lol).

Is there a way I can go about adding a page delete and sync up pdfrx with PDFium's changed document short of modifying the pdfrx source and rebuilding everything? I've been playing around with multiple approaches and can't seem to get that modified (in memory) document back for use with pdfrx.

**Comment by espresso3389** [comment]
Basically, pdfrx's structure is designed only for viewer.
So, the current architecture does not assume the document is updated by any means.

Explaining in more implementation specific things, pdf_pdfium.dart is the actual implementation of PdfDocument and it initializes page array on its first initialization. If you change something of the document after PdfDocument initialization, the change is not propagated to the PdfDocument instance.

So, in the current moment, the only possible solution is to recreate PdfDocument/PdfViewer from the modified file or data.

**Reply by tc4u** [reply]
Thank you so much for the explanation and clarification. I kind of figured there wasn't an easy way to use the PdfViewer as-is in this scenario.

With that said, I took a slightly different approach to work around this by creating a variation of the thumbnail example, which iteratively renders pages. I simply added 'cut' page indices to a list and the rendering for a page is skipped if the current page index is in the cut list. So right now it is a presentation only  removal of the pages.

Now I need to figure out how to write the modified PDF with the removed pages (in Dart).
